src/VisualStudio/IntegrationTest/IntegrationTests/AbstractIntegrationTest.cs

```diff
@@ -1,17 +1,20 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using Microsoft.VisualStudio.IntegrationTest.Utilities;
-using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
-using Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess;
 using System;
 using System.Threading;
-using System.Windows.Automation;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
 
 namespace Roslyn.VisualStudio.IntegrationTests
 {
     [CaptureTestName]
     public abstract class AbstractIntegrationTest : IDisposable
     {
+        /// <summary>
+        /// A long timeout used to avoid hangs in tests, where a test failure manifests as an operation never occurring.
+        /// </summary>
+        protected static readonly TimeSpan HangMitigatingTimeout = TimeSpan.FromMinutes(1);
```

Comment:
I love this because it means invoking FAR on this gives you a set of places where we think we were having issues...